src/getViewForSwagger2.ts

```diff
@@ -116,182 +134,154 @@ export function getViewForSwagger2(opts: CodeGenOptions): ViewData{
-                isSecureBasic: secureTypes.indexOf('basic') !== -1,
-                parameters: [],
-                headers: [],
-                successfulResponseType,
-                successfulResponseTypeIsRef,
-                isLatestVersion: false,
-            };
-
-            latestMethodVersion[method.methodName] = Math.max(latestMethodVersion[method.methodName] || 0, intVersion);
-
-            if(method.isSecure && method.isSecureToken) {
-                data.isSecureToken = method.isSecureToken;
-            }
+const getGlobalParams = <T extends Parameter>(api: { [index: string]: HttpOperation | ReadonlyArray<ReadonlyArray<T>>}): ReadonlyArray<T> => fp.compose(
+    fp.compose(fp.filter<T>(identity), fp.map(([_, value]) => first(value))),
+    fp.filter(isParameters),
+    fp.entries
+)(api);
 
-            if(method.isSecure && method.isSecureApiKey) {
-                data.isSecureApiKey = method.isSecureApiKey;
-            }
+const authorizedMethods = ['GET', 'POST', 'PUT', 'DELETE', 'PATCH', 'COPY', 'HEAD', 'OPTIONS', 'LINK', 'UNLINK', 'PURGE', 'LOCK', 'UNLOCK', 'PROPFIND'];
+const isAuthorizedMethod = (method: string) => authorizedMethods.indexOf(method.toUpperCase()) > -1 
 
-            if(method.isSecure && method.isSecureBasic) {
-                data.isSecureBasic = method.isSecureBasic;
-            }
+function getIntVersion(path: string): number {
+    return parseInt(getVersion(path).substr(1));
+}
 
-            const produces = op.produces || swagger.produces;
-            if(produces) {
-                method.headers.push({
-                  name: 'Accept',
-                  value: `'${produces.join(', ')}'`,
-                });
+// TODO: Do some typescript magic with values from authorizedMethods
+type EndpointDescriptor = [string, HttpEndpointDescriptor];
+type HttpEndpointDescriptor = { [index: string]: HttpOperation};
+type HttpOperationEntryWithGlobalParamsAndPath = [string, string, HttpOperation, ReadonlyArray<Parameter>];
+
+const addPathAndGlobalParams = ([path, api]: EndpointDescriptor): HttpOperationEntryWithGlobalParamsAndPath[] => fp.entries(api).map(([httpVerb, httpOperationDescription]): HttpOperationEntryWithGlobalParamsAndPath => [path, httpVerb, httpOperationDescription, getGlobalParams(api)])
+const addGlobalParamsToEntries = (endpointDescriptor: EndpointDescriptor) => addPathAndGlobalParams(endpointDescriptor);
+
+// TODO: Remind me to clean this method :P
+const makeMethodsFromPaths = (data: ViewData, opts: CodeGenOptions, swagger: Swagger): Method[] => fp.flatten(
+    fp.flatten(fp.entries(swagger.paths).map(addGlobalParamsToEntries))
+        .filter(([_path, httpVerb]) => isAuthorizedMethod(httpVerb))
+        // Ignore deprecated endpoints
+        .filter(([_path, _httpVerb, op]) => !op.deprecated)
+        .map(([path, httpVerb, op, globalParams]) => {
+        // TODO: Start of untested security stuff that needs fixing
```

Comment:
Can you create an issue for this and link it?

Reply:
Yep added issue